### PR TITLE
Add logic to process two empty files

### PR DESCRIPTION
**Pull request by gregzealley**

### CODE REVIEW

File: src/main/java/com/gregzealley/namematcherapi/controllers/PostController.java [hunk 1]
```diff
@@ -18,11 +27,24 @@ public PostController(final NameMatcherService nameMatcherService) {
         this.nameMatcherService = nameMatcherService;
     }
 
-    @RequestMapping(value = "/uploadfiles", method = RequestMethod.POST)
-    public String uploadPrimaryFile(@RequestParam(value = "primary_file") MultipartFile primaryFile,
-                                    @RequestParam(value = "secondary_file") MultipartFile secondaryFile) throws IOException {
+    @RequestMapping(value = "/uploadfiles", method = RequestMethod.POST, produces = "text/csv")
+    public void uploadPrimaryFile(@RequestParam(value = "primary_file") MultipartFile primaryFile,
+                                            @RequestParam(value = "secondary_file") MultipartFile secondaryFile,
+                                            HttpServletResponse response) throws IOException, CsvDataTypeMismatchException, CsvRequiredFieldEmptyException {
```
Comment (ciaransweet) [inline src/main/java/com/gregzealley/namematcherapi/controllers/PostController.java]: Some funky tabbing going on?

File: src/main/java/com/gregzealley/namematcherapi/controllers/PostController.java [hunk 2]
```diff
@@ -18,11 +27,24 @@ public PostController(final NameMatcherService nameMatcherService) {
...
 
-        return nameMatcherService.coordinateNameMatching(primaryFile, secondaryFile);
+        List<MatchResult> results = nameMatcherService.coordinateNameMatching(primaryFile, secondaryFile);
+
+        String filename = "match_results.csv";
+        response.setContentType("text/csv");
+        response.setHeader(HttpHeaders.CONTENT_DISPOSITION,"attachment; filename=\"" + filename + "\"");
+
+        StatefulBeanToCsv<MatchResult> writer = new StatefulBeanToCsvBuilder<MatchResult>(response.getWriter())
```
Comment (ciaransweet) [inline src/main/java/com/gregzealley/namematcherapi/controllers/PostController.java:41]: This is nice!

File: src/main/java/com/gregzealley/namematcherapi/controllers/PostController.java [hunk 3]
```diff
@@ -18,11 +27,24 @@ public PostController(final NameMatcherService nameMatcherService) {
...
+                .withQuotechar(CSVWriter.NO_QUOTE_CHARACTER)
+                .withSeparator(CSVWriter.DEFAULT_SEPARATOR)
+                .withOrderedResults(false)
+                .build();
+
+        writer.write(results);
```
Comment (ciaransweet) [inline src/main/java/com/gregzealley/namematcherapi/controllers/PostController.java:47]: Assuming you don't need a `using` or a `writer.close()` here?

**Review by ciaransweet — CHANGES_REQUESTED** [review]

File: src/main/java/com/gregzealley/namematcherapi/controllers/PostController.java (hunk 1 above)
Comment (gregzealley) [reply src/main/java/com/gregzealley/namematcherapi/controllers/PostController.java]: Fixed

File: src/main/java/com/gregzealley/namematcherapi/controllers/PostController.java (hunk 2 above)
Comment (gregzealley) [reply src/main/java/com/gregzealley/namematcherapi/controllers/PostController.java:41]: I'll send you the link :wink:

File: src/main/java/com/gregzealley/namematcherapi/controllers/PostController.java (hunk 3 above)
Comment (gregzealley) [reply src/main/java/com/gregzealley/namematcherapi/controllers/PostController.java:47]: Hmmm no such method. We'll assume now :stuck_out_tongue:

**Pull request merged**